docs/PrismaMigrationTools.md

```diff
@@ -0,0 +1,64 @@
+# Prisma Migration Tools
+
+## `npm prisma:seed`
+
+### What it does
+
+This command seeds the database, which means populating the database with initial data it requires to run.
+It can be dummy data, basic user accounts, a default language, etc.
+
+### Why to use it
+
+While developing the application, we need some data to be present in order for something to appear on the front end to interact with.
+Running this command populates the database with a basic set of data for development purposes.
+
+For more detailed information about [prisma seed](https://www.prisma.io/docs/guides/database/seed-database), click the link.
+
+## `npm prisma:reset`
+
+### What it does
+
+This command drops the database if possible or performs a soft reset if the database cannot be deleted.
+If the database is dropped, it creates a new database with the same name and runs all migrations and seed scripts.
+If `npm prisma:reset` runs without any errors, then the `npm prisma:seed` function is auto-invoked.
+This means you do not need to run the `npm prisma:seed` command again.
+
+### Why to use it
+
+If you need to undo changes you made to the database schema or if there are migration history conflicts, you should run this command.
+
+You use this command to also restore the data to its original state if you sufficiently modified the data via the front end.
+
+For more detailed information about [prisma reset](https://www.prisma.io/docs/concepts/components/prisma-migrate), click the link.
+
+## `npm prisma:migrate`
+
+### What it does
+
+This command enables you to keep your database schema in sync with the prisma schema.
+
+### Why to use it
+
+After you make a change to the prisma schema, you should run this function to update the database schema.
+There will be a `name` field that the terminal will ask you to enter for the migration.
+Enter a concise and appropriate name that reflects the changes you made to the prisma schema.
+
+Each time you run this command, if there are new changes, a SQL file will be generated that is added to the `migrations` folder.
+Otherwise, `npm prisma:migrate` will run the existing SQL files in the database schema and not require a `name` field to be entered.
+
+For more detailed information about [prisma migrate](https://www.prisma.io/docs/concepts/components/prisma-migrate), click the link.
+
+## `npm prisma:generate`
+
+### What it does
+
+This command generates and initializes the prisma client.
+The prisma client is an auto-generated library that allows us to access data in our application.
+
+Note: The command `npm install` will automatically run `npm prisma:generate`.
```

Comment:
do you think we should add a note about why we set it up this way? I think it's basically to ensure the prisma client is always kept up to date, given that we've also tied the utils package re-generation to npm install. now that I think about it, perhaps these two repository generation items should be de-coupled from npm install since npm install typically fetches remote dependent packages, something we don't necessarily want to do so often. hmmm.

Reply:
I think having the two items coupled with npm install is fine, as npm install is more of a 'catch-all', and the first thing we do when something is not going right with the coding. Instead, we might want to add documentation that explains why and when to run the utils package re-generation and prisma generate so that people can run those separately rather than through `npm install`.

Reply:
Or maybe create a different function that runs both `npm prisma generate` and the utils package regeneration so that if that's all that the user has to run, they could just run that rather than through `npm install`.

Reply:
Yep! My thinking exactly: make a separate npm script in the package.json to run just the prisma generate and utils package regeneration scripts so that devs can run that instead of always running npm install. And yes, documentation to explain those would be really good too. Can you create a separate ticket for this @jinhyunh?